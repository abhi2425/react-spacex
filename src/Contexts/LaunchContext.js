import { axios } from '../Utils/url'
import React, {
  createContext,
  useCallback,
  useContext,
  useEffect,
  useMemo,
  useState,
} from 'react'
import window from 'global/window'

const LaunchContext = createContext()

export const LaunchContextProvider = ({ children }) => {
  const [allSpaceShips, setAllSpaceShips] = useState([])
  const [loading, setLoading] = useState(true)
  const [screenWidth, setScreenWidth] = useState(window.innerWidth)
  const [url, setUrl] = useState('')

  const getAllSpaceShips = useCallback(async (url = 'launches?limit=100') => {
    try {
      setLoading(true)
      const { data } = await axios.get(url)
      if (data) {
        setLoading(false)
        setAllSpaceShips(data)
      }
    } catch (error) {
      setLoading(false)
      console.log(error)
    }
  }, [])

  useEffect(() => {
    if (url) {
      getAllSpaceShips(url)
      window.history.replaceState(null, 'SpaceX Launch Program', url)
    }
  }, [url, getAllSpaceShips])

  const values = useMemo(
    () => ({
      getAllSpaceShips,
      setScreenWidth,
      setUrl,
      allSpaceShips,
      loading,
      screenWidth,
    }),
    // eslint-disable-next-line react-hooks/exhaustive-deps
    [allSpaceShips, loading, screenWidth]
  )

  return (
    <LaunchContext.Provider value={values}>{children}</LaunchContext.Provider>
  )
}

export const useLaunchContext = () => useContext(LaunchContext)
