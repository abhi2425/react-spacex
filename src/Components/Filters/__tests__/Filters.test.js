import React from 'react'
import ReactDOM from 'react-dom'
import Filter from '../Filters'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'
import { LaunchContextProvider } from '../../../Contexts/LaunchContext'
afterEach(cleanup)
it('should render Filter Component', () => {
  const div = document.createElement('div')

  ReactDOM.render(
    <LaunchContextProvider>
      <Filter />
    </LaunchContextProvider>,
    div
  )
  ReactDOM.unmountComponentAtNode(div)

  const { getByTestId } = render(
    <LaunchContextProvider>
      <Filter />
    </LaunchContextProvider>
  )
  expect(getByTestId('filter')).toHaveClass('filters')
})

it('should match snapshot', () => {
  const tree = renderer.create(
    <LaunchContextProvider>
      <Filter />
    </LaunchContextProvider>
  )
  expect(tree).toMatchSnapshot()
})
