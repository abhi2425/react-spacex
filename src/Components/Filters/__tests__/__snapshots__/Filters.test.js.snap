// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<section
  className="filters"
  data-testid="filter"
  style={
    Object {
      "marginLeft": "1.5rem",
      "position": "fixed",
    }
  }
>
  <h4>
    Filters
  </h4>
  <p>
    Launch Year 
  </p>
  <div
    className="filter-container"
  >
    <button
      className="filter-button "
      data-testid="yearFilter"
      onClick={[Function]}
    >
      2006
    </button>
    <button
      className="filter-button "
      data-testid="yearFilter"
      onClick={[Function]}
    >
      2007
    </button>
    <button
      className="filter-button "
      data-testid="yearFilter"
      onClick={[Function]}
    >
      2008
    </button>
    <button
      className="filter-button "
      data-testid="yearFilter"
      onClick={[Function]}
    >
      2009
    </button>
    <button
      className="filter-button "
      data-testid="yearFilter"
      onClick={[Function]}
    >
      2010
    </button>
    <button
      className="filter-button "
      data-testid="yearFilter"
      onClick={[Function]}
    >
      2011
    </button>
    <button
      className="filter-button "
      data-testid="yearFilter"
      onClick={[Function]}
    >
      2012
    </button>
    <button
      className="filter-button "
      data-testid="yearFilter"
      onClick={[Function]}
    >
      2013
    </button>
    <button
      className="filter-button "
      data-testid="yearFilter"
      onClick={[Function]}
    >
      2014
    </button>
    <button
      className="filter-button "
      data-testid="yearFilter"
      onClick={[Function]}
    >
      2015
    </button>
    <button
      className="filter-button "
      data-testid="yearFilter"
      onClick={[Function]}
    >
      2016
    </button>
    <button
      className="filter-button "
      data-testid="yearFilter"
      onClick={[Function]}
    >
      2017
    </button>
    <button
      className="filter-button "
      data-testid="yearFilter"
      onClick={[Function]}
    >
      2018
    </button>
    <button
      className="filter-button "
      data-testid="yearFilter"
      onClick={[Function]}
    >
      2019
    </button>
    <button
      className="filter-button "
      data-testid="yearFilter"
      onClick={[Function]}
    >
      2020
    </button>
  </div>
  <p>
    Successful Launch 
  </p>
  <div
    className="filter-container"
  >
    <button
      className="filter-button false"
      onClick={[Function]}
    >
      true
    </button>
    <button
      className="filter-button false"
      onClick={[Function]}
    >
      false
    </button>
  </div>
  <p>
    Successful Landing 
  </p>
  <div
    className="filter-container"
  >
    <button
      className="filter-button false"
      onClick={[Function]}
    >
      true
    </button>
    <button
      className="filter-button false"
      onClick={[Function]}
    >
      false
    </button>
  </div>
</section>
`;
